Tests/ClientTests/Mocks/MockNavigationController.swift

```diff
@@ -0,0 +1,51 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+@testable import Client
+
+class MockNavigationController: UIViewController, NavigationController {
+    var viewControllers: [UIViewController] = []
```

Comment:
I had the impression we introduced `NavigationController` protocol to mock easily without needing `UIKit`. I'm curious why does it need to inherit from `UIViewController`.

Reply:
I wanted to get for free the `dismiss` and `present` method from UIViewController on the `NavigationController`, so I don't have to declare them. I can remove the conformance to UIViewController if you think it's not beneficial.

Reply:
Trying to remove it, it makes it harder to test since we don't have the Equatable from view controller available anymore.

Reply:
We wouldn't want any class other than a UIViewController or it's subclasses to conform to NavigationController, it wouldn't make sense so I think it's fine to keep it. 
The purpose of the mock isn't to avoid UIKit (although that is a nice side effect when possible) the mock is to control for inputs and check for outputs in the unit tests without having to work around the implementation details of the real instance.

Reply:
I guess my question now is why do we need `NavigationController` protocol? Because `NavigationController` + `UIViewController` = `UINavigationController` so we might as well use that.

Reply:
To have mocks for tests. We don't have much control over `UINavigationController`